test/unity/toolkits/drawing_classifier/test_dc_prediction.cxx

```diff
@@ -225,25 +236,19 @@ void prediction_test_driver(size_t batch_size, size_t num_of_rows,
     class_labels.push_back(std::to_string(ii));
   }
 
-  const std::string feature_name = "feature";
-  const std::string target_name = "target";
-
   // name 'target', 'feature' are used by create_iterator
-  drawing_data_generator data_generator(/* is_bitmap_based */ true,
+  drawing_data_generator data_generator(/* is_bitmap_based */ is_bitmap_based,
                                         num_of_rows, class_labels, target_name,
                                         feature_name);
 
   gl_sframe my_data = data_generator.get_data();
   TS_ASSERT_EQUALS(my_data.size(), num_of_rows);
 
-  mock_model.create_iterator_calls_.push_back([=](data_iterator::parameters) {
-    data_iterator::parameters my_params;
-    my_params.data = my_data;
-    my_params.is_train = false;
-    my_params.feature_column_name = feature_name;
-    my_params.target_column_name = target_name;
-    return std::unique_ptr<data_iterator>(new simple_data_iterator(my_params));
-  });
+  mock_model.create_iterator_calls_.push_back(
+      [=](data_iterator::parameters my_params) {
```

Comment:
Why remove lines for setting up params? I intend to explicitly set params with specific values for test purposes.

Reply:
Because `my_data` here can be stroke-based data, whereas the data iterator can only take in bitmap-based data.

Since we wrote this code with an intention to use the real data iterator, we would have to set up the data (by calling the real `create_iterator` where the stroke to bitmap conversion happens) appropriately for consumption by the data iterator.

Reply:
It's very strange for the callback to set the values for `my_params`. For mocks, these callbacks are intended to be the (mock) implementation, and to make assertions about what the parameters are.

For example, it would be very weird if `drawing_classifier::create_iterator` blindly overwrote the parameters you pass it as an argument

Reply:
That's by design to override the `params`. There's no way to override inside of predict call. This is designed to bypass the infinite loop bug in dc. And also pass extra column name info into params. I don't remember why I need this workaround. You can run the test to see if it passes. If it passed, then you are good to go.